docker/coverage/Dockerfile

```diff
@@ -0,0 +1,18 @@
+# base image that creates all necessary dependencies for
```

Comment:
Ok, I have to admit, I'm not a big fan of creating separate Buildfiles for each of these targets especially putting them in a separate directory.  Everytime we added in a new python module dependency, we would have to come down into this directory as well as add them in.  It won't be top of mind to people.

I'm going to suggest something a little more funky.  We add in a `--coverage` option to the `build_package` command.  It only works for docker/k8s builds.  When that is specified, before it uses the build file, it first rewrites it to add in the necessary changes.  From what I can see, you just need to add in `coverage` to the `RUN pip` line.. and then replace the `CMD` line.  Correct?  You could write a little regex that catches those lines and modifies them.  You could also fail with a big error message if you do not find those lines to modify.

That would also mean you could revert your changes to `build_package.py` to take in a different docker file path.

What do you think?  I know it's a little weird... but it will create less maintenance headaches in the future.